### PR TITLE
Update p-stepped-list__title to allow child elements

### LINKED ISSUE #2518: .p-stepped-list__title wraps incorrectly if it contains child elements
**Describe the bug**
If a `p-stepped-list__title` contains another HTML element, such as a `<strong>` tag, space either side of the element collapses, and the element will wrap incorrectly at different device widths.

**To Reproduce**
1. Go to https://codepen.io/sowasred2012/pen/wvwevwN?editors=1000
2. Resize the browser

**Expected behavior**
Child elements of the `p-stepped-list__title` should not deform their parent element.

**Screenshots**
Screenshot 2019-08-28 at 15 38 27

### PR DESCRIPTION
**Pull request by sowasred2012**
## Done

Changed p-stepped-list__title to not use `display: flex;` so child elements don't cause layout problems

Fixes #2518 

## QA

- Open [demo](https://vanilla-framework-3694.demos.haus/docs/examples/patterns/lists/lists-stepped)
- Reduce the width of your browser until the fourth item "Design and Build" begins to wrap. It should wrap as expected, unlike https://codepen.io/sowasred2012/pen/xxgXvQg

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [component status page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/component-status.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

File: scss/_patterns_lists.scss [hunk 1]
```diff
@@ -260,6 +260,12 @@ $spv-list-item--inner: null;
...
 
     h#{$i}.p-stepped-list__title {
+      padding-left: $bullet-width-mobile + $sph-inner;
+
+      @media (min-width: $breakpoint-heading-threshold) {
+        padding-left: $bullet-width + $sph-inner;
+      }
+
       &::before {
         flex-shrink: 0;
```
Comment (bartaz) [inline scss/_patterns_lists.scss:270]: I guess we can remove that as well now that `flex` is gone from parent.

Comment (sowasred2012) [reply scss/_patterns_lists.scss:270]: true!

Comment (sowasred2012) [reply scss/_patterns_lists.scss]: @bartaz updated

**Review by bartaz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**